index.js

```diff
@@ -355,15 +355,11 @@ function Physics (mcData, world) {
       if (isOnLadder(world, pos)) {
         vel.x = math.clamp(-physics.ladderMaxSpeed, vel.x, physics.ladderMaxSpeed)
         vel.z = math.clamp(-physics.ladderMaxSpeed, vel.z, physics.ladderMaxSpeed)
-        vel.y = Math.max(vel.y, entity.control.sneak ? 0 : -physics.ladderMaxSpeed)
+        vel.y = Math.max(entity.control.jump ? physics.ladderClimbSpeed : vel.y, entity.control.sneak ? 0 : -physics.ladderMaxSpeed)
       }
 
       moveEntity(entity, world, vel.x, vel.y, vel.z)
 
-      if (entity.isCollidedHorizontally && isOnLadder(world, pos)) {
-        vel.y = physics.ladderClimbSpeed // climb ladder
```

Comment:
See previous comment. Corresponding code is line 223 of https://www.mcpk.wiki/wiki/SourceCode:EntityLivingBase

Reply:
I've tested in 1.16 that checking whether entity occupies same block of a climbable block is enough to be able to climb, at this point the entity is not colliding with anything except the ground.